backend/Origam.DA.Service/AbstractSqlDataService.cs

```diff
@@ -1723,400 +1844,531 @@ private List<ColumnData> GetAllQueryColumns(DataStructureQuery query)
-                            SchemaDbCompareResult result = new SchemaDbCompareResult
-                            {
-                                ResultType = DbCompareResultType.ExistingButDifferent,
-                                ItemName = key,
-                                SchemaItem = fld,
-                                Remark = differenceDescription,
-                                SchemaItemType = typeof(FieldMappingItem),
-                                Script = gen.AlterColumnDdl(fld)
-                            };
-                            results.Add(result);
-                        }
-                    }
-                    else
-                    {
-                        // does not exist in schema
-                        SchemaDbCompareResult result = new SchemaDbCompareResult();
-                        result.ResultType = DbCompareResultType.MissingInSchema;
-                        result.ItemName = key;
-                        result.SchemaItem = null;
-                        result.ParentSchemaItem = schemaTableList[row["TABLE_NAME"]];
-                        result.SchemaItemType = typeof(FieldMappingItem);
-                        results.Add(result);
-                    }
+	                continue;
+                }
+                if(schemaColumnList.ContainsKey(key))
+                {
+	                var fieldMappingItem = schemaColumnList[key] 
+		                as FieldMappingItem;
+	                var differenceDescription = "";
+	                if(((string)row["IS_NULLABLE"] == "YES")
+	                && !fieldMappingItem.AllowNulls)
+	                {
+		                differenceDescription =
+			                (differenceDescription == "" ? "" : "; ")
+			                + "AllowNulls: Schema-NO, Database-YES";
+	                }
+	                if(((string)row["IS_NULLABLE"] == "NO") 
+	                && fieldMappingItem.AllowNulls)
+	                {
+		                differenceDescription =
+			                (differenceDescription == "" ? "" : "; ")
+			                + "AllowNulls: Schema-YES, Database-NO";
+	                }
+	                if(CompareType(row, abstractSqlCommandGenerator
+		                   .DdlDataType(fieldMappingItem.DataType, 
+			                   fieldMappingItem.MappedDataType).ToUpper()))
+	                {
+		                differenceDescription 
+			                = (differenceDescription == "" ? "" : "; ")
+			                  + "DataType: Schema-"
+			                  + abstractSqlCommandGenerator.DdlDataType(
+				                  fieldMappingItem.DataType, 
+				                  fieldMappingItem.MappedDataType)
+			                  + ", Database-" + (string)row["DATA_TYPE"];
+	                }
+	                if((fieldMappingItem.DataType == OrigamDataType.String)
+	                   && !row.IsNull("CHARACTER_MAXIMUM_LENGTH")
+	                   && ((int)row["CHARACTER_MAXIMUM_LENGTH"] 
```

Comment:
I think It looks more readbale to add an indentation before "&&" like you did here. But this seems to be an exception there are many if statements with the indents befere the new line missing.

Reply:
The rest of the code is adjusted accordingly.